flocker/volume/functional/test_filesystems_zfs.py

```diff
@@ -541,7 +540,9 @@ def created(filesystem):
             path = filesystem.get_path()
             # Try to write more than 64MB of data.
             with path.child(b"ok").open("w") as fObj:
-                self.assertRaises(
-                    IOError, write_flush, fObj, b"x" * 64 * 1024 * 1024)
+                exc = self.assertRaises(
+                    IOError, write_flush, fObj, b"x" * (maximum_size + 1))
```

Comment:
This creates a 40 MiB in-memory string. Is it possible to do this with a loop that writes a smaller byte string multiple times?

If so, this would be a place to use the `with` context manager:
```python
with self.assertRaises(IOError) as exc:
    bufsize = 8192
    buf = 'x' * bufsize
    for i in range(maximum_size / bufsize + 1):
        write_flush(fObj, buf)
```